src/model/documentselection.js

```diff
@@ -707,9 +707,15 @@ function clearAttributesStoredInElement( changes, batch ) {
 		return;
 	}
 
-	const storedAttributes = Array.from( changeParent.getAttributeKeys() ).filter( key => key.startsWith( storePrefix ) );
+	const anyStoredAttributes = Array.from( changeParent.getAttributeKeys() ).some( key => key.startsWith( storePrefix ) );
```

Comment:
Damn... I haven't even thought about enqChanges not being synchronous ;<

Regarding empty enqChanges – I can't think of any issue that would cause, so I guess we can remove that `if()`.